test/unit/lodash.spec.js

```diff
@@ -0,0 +1,275 @@
+            var params = datasets.every2();
+            var array = params[0];
+            expect(array.length).to.eql(4);
+            expect(_.every.apply(_, datasets.every2())).to.eql(true);
+        });
+    });
+
+    describe('filter', function () {
+        it('should return filtered array with 2 objects with matching predicate of age', function () {
+            var params = datasets.filter();
+            var array = params[0];
+            expect(array.length).to.eql(5);
+            expect(_.filter.apply(_, datasets.filter())).to.eql([{name: 'Ali', age: 33, active: false},
+                {name: 'Kali', age: 21, active: true}]);
+        });
+        it('should return filtered array with 3 objects and match for age and active', function () {
+            var params = datasets.filter2();
+            var array = params[0];
+            expect(array.length).to.eql(5);
+            expect(_.filter.apply(_, datasets.filter2())).to.eql([{name: 'Bow', age: 23, active: true},
+                {name: 'Row', age: 23, active: true},
+                {name: 'Sou', age: 23, active: true}]);
+        });
+
+    });
+
+    describe('find', function () {
+        it('should find Helen', function () {
+            var testArray = _.fill(new Array(50), 1).map(function () {
+                if (Math.random() < 0.2) {
+                    return {name: 'Kodi', age: 12, active: true};
+                } else if (Math.random() < 0.4) {
+                    return {name: 'Olaf', age: 32, active: false};
+                } else if (Math.random() < 0.6) {
+                    return {name: 'Helen', age: 65, active: true};
+                } else if (Math.random() < 0.8) {
+                    return {name: 'Pluto', age: 25, active: false};
+                } else {
+                    return {name: 'Rudi', age: 43, active: true};
+                }
+            });
+            var params = datasets.find(testArray);
+            var array = params[0];
+            var method = params[1];
+            var result = _.find.apply(_, params);
+            expect(array.length).to.eql(50);
+            expect(Object.keys(method).length).to.eql(1);
+            expect(_.values(result)).to.contain('Helen');
+            expect(_.values(result)).to.contain(65);
+        });
+        it('should return first matched element for name property', function () {
+            var params = datasets.find2();
+            var array = params[0];
+            expect(array.length).to.eql(3);
+            array.forEach(function (n) {
+                expect(_.has(n, 'NAME')).to.eql(true);
+                expect(_.has(n, 'active')).to.eql(true);
+            });
+            expect(_.find.apply(_, datasets.find2())).to.eql({
+                NAME: 'Rudi',
+                age: 15,
+                active: true,
+                gender: 'M'
+            });
+        });
+    });
+
+    describe('groupBy', function () {
```

Comment:
groupBy: Tests here are a little bit better but still I was able to avoid using indicated math functions and have answer that doesn't meet conditions(and works). Answers are available straight from documentation and tasks are not innovative in any way